coaster/sqlalchemy.py

```diff
@@ -152,25 +214,106 @@ def __repr__(self):
         return '<%s %s>' % (self.__class__.__name__, self.id)
 
 
+class UuidMixin(object):
+    """
+    Provides a ``uuid`` attribute that is either a SQL UUID column or an alias
+    to the existing ``id`` column if the class uses UUID primary keys. Also
+    provides hybrid properties ``url_id``, ``buid`` and ``suuid`` that provide
+    hex, BUID and ShortUUID representations of the ``uuid`` column.
+
+    :class:`UuidMixin` must appear before other classes in the base class order::
+
+        class MyDocument(UuidMixin, BaseMixin, db.Model):
+            pass
+
+    Compatibility table:
+
+    +-----------------------+-------------+-----------------------------------------+
+    | Base class            | Compatible? | Notes                                   |
+    +=======================+=============+=========================================+
+    | BaseMixin             | Yes         |                                         |
+    +-----------------------+-------------+-----------------------------------------+
+    | BaseIdNameMixin       | Yes         |                                         |
+    +-----------------------+-------------+-----------------------------------------+
+    | BaseNameMixin         | N/A         | ``name`` is secondary key, not ``uuid`` |
+    +-----------------------+-------------+-----------------------------------------+
+    | BaseScopedNameMixin   | N/A         | ``name`` is secondary key, not ``uuid`` |
+    +-----------------------+-------------+-----------------------------------------+
+    | BaseScopedIdMixin     | No          | Conflicting :attr:`url_id` attribute    |
+    +-----------------------+-------------+-----------------------------------------+
+    | BaseScopedIdNameMixin | No          | Conflicting :attr:`url_id` attribute    |
+    +-----------------------+-------------+-----------------------------------------+
+    """
+    @declared_attr
+    def uuid(cls):
+        """UUID column, or synonym to existing :attr:`id` column if that is a UUID"""
+        if hasattr(cls, '__uuid_primary_key__') and cls.__uuid_primary_key__:
+            return synonym('id')
+        else:
+            return Column(UUIDType(binary=False), default=uuid_.uuid4, unique=True)
+
+    @hybrid_property
+    def url_id(self):
+        """URL-friendly UUID representation as a hex string"""
+        return self.uuid.hex
```

Comment:
Apps can choose to expose `suuid` in URLs by (a) deriving from `UuidMixin` and using `suuid` or `url_name_suuid` in URL tags.